core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -241,6 +241,12 @@ FUNC _start , :
 UNWIND(	.fnstart)
 UNWIND(	.cantunwind)
 
+	/*
+	 * 32bit entry is expected to execute Supervisor mode,
+	 * some bootloader may enter in Supervisor or Monitor
+	 */
+	cps	#CPSR_MODE_SVC
```

Comment:
The can't be done before `bootargs_entry` as one parameter is passed in `lr`

Reply:
hmm... yes... and no. well, i'll shift it afterward....